crates/host/src/wasmbus/mod.rs

```diff
@@ -326,9 +343,78 @@ impl Invocation {
             msg,
             id,
             encoded_claims,
-            host_id,
+            host_id: host_id.to_string(),
         })
     }
+
+    /// A fully-qualified URL indicating the origin of the invocation
+    pub fn origin_url(&self) -> String {
+        self.origin.url()
+    }
+
+    /// A fully-qualified URL indicating the target of the invocation
+    pub fn target_url(&self) -> String {
+        format!("{}/{}", self.target.url(), self.operation)
+    }
+
+    /// The hash of the invocation's target, origin, and raw bytes
+    pub fn hash(&self) -> String {
+        invocation_hash(
+            self.target_url(),
+            self.origin_url(),
+            &self.operation,
+            &self.msg,
+        )
+    }
+
+    /// Validates the current invocation to ensure that the invocation claims have
+    /// not been forged, are not expired, etc
+    pub fn validate_antiforgery(&self, valid_issuers: &[String]) -> Result<()> {
+        if !self.host_id.starts_with('N') || self.host_id.len() != 56 {
```

Comment:
Can we store a `KeyPair` or `Arc<KeyPair>` to remove the need for this check?

Reply:
I don't think we should worry about storing a KeyPair, but I will change this check to parse a KeyPair instead of the naive check